Tests/SymmetricKeyTests.swift

```diff
@@ -0,0 +1,130 @@
+
+        let key = Data(bytes: [
+            0x19, 0xac, 0x20, 0x82, 0xe1, 0x72, 0x1a, 0xb5, 0x8a, 0x6a, 0xfe, 0xc0, 0x5f, 0x85, 0x4a, 0x52
+        ])
+
+        let jwk = SymmetricKey(
+            key: key,
+            additionalParameters: [ "alg": SymmetricKeyAlgorithm.A256CBCHS512.rawValue ]
+        )
+
+        XCTAssertEqual(jwk.key, "GawgguFyGrWKav7AX4VKUg")
+        XCTAssertEqual(jwk.keyType, .SYM)
+        XCTAssertEqual(jwk["alg"], "A256CBC-HS512")
+
+        XCTAssertEqual(
+            "{\"kty\":\"oct\",\"alg\":\"A256CBC-HS512\",\"k\":\"GawgguFyGrWKav7AX4VKUg\"}",
+            jwk.jsonString()!
+        )
+    }
+
+    func testParsingSymmetricKeyFromJSONData() {
+        let key = Data(bytes: [
+            0x19, 0xac, 0x20, 0x82, 0xe1, 0x72, 0x1a, 0xb5, 0x8a, 0x6a, 0xfe, 0xc0, 0x5f, 0x85, 0x4a, 0x52
+            ])
+
+        let json = SymmetricKey(
+            key: key,
+            additionalParameters: [ "alg": SymmetricKeyAlgorithm.A256CBCHS512.rawValue ]
+        ).jsonData()!
+
+
+        let jwk = try! SymmetricKey(data: json)
+
+        XCTAssertEqual(jwk.key, "GawgguFyGrWKav7AX4VKUg")
+        XCTAssertEqual(jwk.keyType, .SYM)
+        XCTAssertEqual(jwk["alg"], "A256CBC-HS512")
+
+        XCTAssertEqual(
+            "{\"kty\":\"oct\",\"alg\":\"A256CBC-HS512\",\"k\":\"GawgguFyGrWKav7AX4VKUg\"}",
+            jwk.jsonString()!
+        )
+    }
+
+    func testParsingSymmetricKeyFromOtherKeyRepresentation() {
+        let key: ExpressibleAsSymmetricKeyComponents = Data(bytes: [
+            0x19, 0xac, 0x20, 0x82, 0xe1, 0x72, 0x1a, 0xb5, 0x8a, 0x6a, 0xfe, 0xc0, 0x5f, 0x85, 0x4a, 0x52
+            ])
+
+        let json = try! SymmetricKey(
+            key: key,
+            additionalParameters: [ "alg": SymmetricKeyAlgorithm.A256CBCHS512.rawValue ]
+            ).jsonData()!
+
+
+        let jwk = try! SymmetricKey(data: json)
+
+        XCTAssertEqual(jwk.key, "GawgguFyGrWKav7AX4VKUg")
+        XCTAssertEqual(jwk.keyType, .SYM)
+        XCTAssertEqual(jwk["alg"], "A256CBC-HS512")
+
+        XCTAssertEqual(
+            "{\"kty\":\"oct\",\"alg\":\"A256CBC-HS512\",\"k\":\"GawgguFyGrWKav7AX4VKUg\"}",
+            jwk.jsonString()!
+        )
+    }
+
+    func testSymmetricKeyToData() {
+        let key = Data(bytes: [
+            0x19, 0xac, 0x20, 0x82, 0xe1, 0x72, 0x1a, 0xb5, 0x8a, 0x6a, 0xfe, 0xc0, 0x5f, 0x85, 0x4a, 0x52
+            ])
+
+        let jwk = SymmetricKey(key: key)
+
+        let keyData = try! jwk.converted(to: Data.self)
+
+        XCTAssertEqual(keyData, key)
+    }
+
+    func testMalformedSymmetricKeyToData() {
+        let json = "{\"kty\":\"oct\",\"alg\":\"A256CBC-HS512\",\"k\":\"+++==notbase64url==---\"}".data(using: .utf8)!
+
+        XCTAssertThrowsError(try SymmetricKey(data: json))
+    }
+
+    func testDecodingFromJSONWithMissingKeyType() {
+        let json = "{\"alg\":\"A256CBC-HS512\",\"k\":\"+++==notbase64url==---\"}".data(using: .utf8)!
```

Comment:
Maybe use a correct key otherwise you don't know if it throws because of the missing keytype or the not base64URL key.

Reply:
Good catch!